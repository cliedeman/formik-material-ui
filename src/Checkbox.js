/* @flow */
import Checkbox from '@material-ui/core/Checkbox';

import createComponent from './createComponent';

export default createComponent(
  Checkbox,
  ({ field, form: { isSubmitting }, disabled = false, ...props }) => ({
    disabled: isSubmitting || disabled,
    ...props,
    ...field,
    // TODO: is this the correct way?
    value: field.value ? 'checked' : '',
  })
);
