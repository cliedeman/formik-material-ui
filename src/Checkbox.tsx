// @ts-ignore
import * as React from 'react';
import MuiCheckbox, {
  CheckboxProps as MuiCheckboxProps,
} from '@material-ui/core/Checkbox';
import { FieldProps } from 'formik';
import { Omit } from './types';

export interface CheckboxProps extends FieldProps, Omit<MuiCheckboxProps, 'form'> {}

const Checkbox: React.ComponentType<CheckboxProps> = ({
  field,
  form: { isSubmitting },
  disabled = false,
  ...props
}) => (
  <MuiCheckbox
    disabled={isSubmitting || disabled}
    {...props}
    {...field}
    // TODO: is this the correct way?
    value={field.value ? 'checked' : ''}
  />
);

Checkbox.displayName = 'FormikMaterialUICheckbox';

export default Checkbox;
