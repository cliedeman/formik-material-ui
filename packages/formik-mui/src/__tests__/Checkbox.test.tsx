import React from 'react';
import '@testing-library/jest-dom/vitest';
import { test, expect, vi } from 'vitest';
import { Field } from 'formik';

import { render, fireEvent } from './utils';
import { Checkbox } from '../Checkbox';

test('renders', async () => {
  const onSubmit = vi.fn();
  const { asFragment } = render(
    <Field component={Checkbox} name="checked" type="checkbox" />,
    {
      initialValues: {
        checked: false,
      },
      onSubmit,
    }
  );

  expect(asFragment()).toMatchSnapshot();
});

test('checked', async () => {
  // given
  const onSubmit = vi.fn();
  const { getByTestId, findByText } = render(
    <Field
      component={Checkbox}
      name="checked"
      label="Checkbox"
      type="checkbox"
      inputProps={{
        'data-testid': 'checkbox',
      }}
    />,
    {
      onSubmit,
      initialValues: {
        checked: false,
      },
    }
  );

  const input = getByTestId('checkbox');
  const submit = getByTestId('submit');

  // when
  fireEvent.change(input, { target: { value: 'checked' } });
  fireEvent.click(submit);

  await findByText('submitted');

  // then
  expect(onSubmit).toBeCalledWith(
    {
      checked: false,
    },
    expect.anything()
  );
});
