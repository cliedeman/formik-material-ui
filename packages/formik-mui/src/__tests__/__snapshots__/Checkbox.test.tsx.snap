// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders 1`] = `
<DocumentFragment>
  <form
    action="#"
    data-testid="form"
  >
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
    >
      <input
        class="PrivateSwitchBase-input css-1m9pwf3"
        data-indeterminate="false"
        name="checked"
        type="checkbox"
        value="false"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CheckBoxOutlineBlankIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </span>
    <button
      data-testid="submit"
    >
      submit
    </button>
  </form>
</DocumentFragment>
`;
